### PR TITLE
[Bug] extractProgramArgs parsing args parameter error

### PR DESCRIPTION
**Pull request by fantasticKe**
…arameter error

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/streampark/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

### CODE REVIEW

**Comment by wolfboys** [comment]
It looks good, can you provide test case?

**Comment by fantasticKe** [comment]
This is my test case, I've tried it in streamPark and it works as expected
image

**Comment by fantasticKe** [comment]
Ok, I have added
*fantasticMao*
--------------------------------------------------------------------------------------



*我们都是时间的旅人，在每一个人生阶段旅行，在每一片回忆的风景里，努力修补自己。所以，不要人云亦云地活着，每个人都有属于自己的路，谁也替不了谁。*
--------------------------------------------------------------------------------------


benjobs ***@***.***> 于2022年9月19日周一 22:10写道：

**Comment by wolfboys** [comment]
> This is my test case, I've tried it in streamPark and it works as expected image

oh, You need to provide code of the test case. Please refer to the latest code, which is located in streampark-flink-submit-core/ParameterTestCase , you can write the test code in this class.

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala [hunk 1]
```diff
@@ -356,8 +358,12 @@ trait FlinkSubmitTrait extends Logger {
 
     val programArgs = new ArrayBuffer[String]()
 
-    Try(submitRequest.args.split("\\s+")).getOrElse(Array()).foreach(x => if (x.nonEmpty) programArgs += x)
-
+    //    Try(submitRequest.args.split("\\s+")).getOrElse(Array()).foreach(x => if (x.nonEmpty) programArgs += x)
+    val pattern = Pattern.compile("[^\\s\"']+|\"[^\"]*\"|'[^']*'")
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: The Reg-exp Pattern  need to be defined in advance. Please refer to line 34 of streampark-flink-submit-api/FlinkSubmitter

**Comment by fantasticKe** [comment]
okay, I have added the corresponding test case. pls check it out.
*fantasticMao*
--------------------------------------------------------------------------------------



*我们都是时间的旅人，在每一个人生阶段旅行，在每一片回忆的风景里，努力修补自己。所以，不要人云亦云地活着，每个人都有属于自己的路，谁也替不了谁。*
--------------------------------------------------------------------------------------


benjobs ***@***.***> 于2022年9月20日周二 10:38写道：

**Comment by wolfboys** [comment]
> okay, I have added the corresponding test case. pls check it out. *fantasticMao* ------------------------------------------

some conflicts need to be resolved.

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/test/scala/org/apache/streampark/flink/submit/test/ParameterTestCase.scala [hunk 2]
```diff
@@ -39,7 +43,31 @@ class ParameterTestCase {
...
+    Try(argsStr.split("\\s+")).getOrElse(Array()).foreach(x => if (x.nonEmpty) oldProgramArgs += x)
+    println(oldProgramArgs)
+    //new
+    val newProgramArgs = new ArrayBuffer[String]()
+    val pattern = Pattern.compile("[^\\s\"']+|\"[^\"]*\"|'[^']*'")
+    val regexMatcher = pattern.matcher(argsStr)
+    while (regexMatcher.find()) {
+      newProgramArgs += regexMatcher.group().replaceAll("\"", "")
+    }
+    println(newProgramArgs)
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/test/scala/org/apache/streampark/flink/submit/test/ParameterTestCase.scala]: use Assertions.assertEquals determine whether it meets expectations,not println.

e.g:
```
Assertions.assertEquals("--ckUrl", newProgramArgs(7))
Assertions.assertEquals("jdbc:clickhouse://localhost:8123/default", newProgramArgs(8))
...
```

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala [hunk 3]
```diff
@@ -356,8 +359,12 @@ trait FlinkSubmitTrait extends Logger {
 
     val programArgs = new ArrayBuffer[String]()
 
-    Try(submitRequest.args.split("\\s+")).getOrElse(Array()).foreach(x => if (x.nonEmpty) programArgs += x)
-
+    //    Try(submitRequest.args.split("\\s+")).getOrElse(Array()).foreach(x => if (x.nonEmpty) programArgs += x)
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: you can remove this line.

**Comment by wolfboys** [comment]
In the case of complex parameters, the code have bug, e.g:

--mypass '123\'333'
--mypass1 "aaa\"bbb"

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**